### PR TITLE
[AIRFLOW-3263] Ignore exception when 'run' tries to kill already killed job

### PR DESCRIPTION
**Pull request by potiuk**
Sometimes when you run tasks from command line you get exit code = 1 due
to race condition (job runner tries to get process group from the
process that has already been terminated in the meantime)

Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-3263
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
Sometimes when you run tasks from command line you get exit code = 1 due
to race condition (job runner tries to get process group from the
process that has already been terminated in the meantime)

### Tests

- [x] My PR  does not need testing for this extremely good reason:
It's rather difficult to simulate such race. It's not really worth It to build test for it with such simple change. 

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
No new functionality

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by ashb** [comment]
Do we need to (can we?) reap child processes to avoid zombie processes?

**Comment by potiuk** [comment]
@ashb I don't think we can - because we would somehow have to know which are those processes and in this particular case the parent has already died, so all the children are automatically moved to have 1 as parent (if they survived), so we have basically no way of knowing who are the children we should kill.

File: airflow/utils/helpers.py [hunk 1]
```diff
@@ -234,7 +234,15 @@ def on_terminate(p):
     children = parent.children(recursive=True)
     children.append(parent)
 
-    log.info("Sending %s to GPID %s", sig, os.getpgid(pid))
+    try:
+        pg = os.getpgid(pid)
+    except OSError as err:
+        # Skip if not such process - we experience a race and it just terminated :)
+        if err.errno == 3:
```
Comment (ashb) [inline airflow/utils/helpers.py]: Worth comparing to `errno.ESRCH` to avoid magic number?

**Review by ashb — APPROVED** [review]
One possible improvement, otherwise looks good

**Comment by potiuk** [comment]
Sure @ashb . Magic is not good :).

**Review by ashb — APPROVED** [review]

**Pull request merged**